src/dev_runner_api.rs

```diff
@@ -43,18 +39,19 @@ pub async fn set_runnable_project(
 }
 
 #[post("/exec-command")]
-pub async fn exec_command(context: web::Data<RwLock<RunnerContext>>, payload: web::Json<Script>)
-    -> web::Json<BasicResponse> {
-
-    let projects = context
-        .read()
-        .unwrap()
-        .projects.clone();
-
-        // command_exec_utils::exec_scripts("dev", projects);
-
-        web::Json(BasicResponse {
-            msg: "run".parse().unwrap(),
-        })
-
+pub async fn exec_command(
+    context: web::Data<RwLock<RunnerContext>>,
+    payload: web::Json<Script>,
+) -> web::Json<BasicResponse> {
+    let active_processes = context.write().unwrap().child_processes.clone();
+    command_exec_utils::terminate_all(&active_processes);
```

Comment:
should fix termination - while it does the termination the process isn't clear on time